client/src/components/_common/Searchbar/Searchbar.jsx

```diff
@@ -0,0 +1,185 @@
+  const location = useLocation();
+  const { query_string: hasQuery, filter: filterType } = queryString.parse(
+    location.search
+  );
+
+  const displayName = useSystemDisplayName({ system, scheme });
+  let sectionName;
+
+  if (siteSearch) {
+    sectionName = 'Site';
+  } else if (scheme === 'projects') {
+    sectionName = 'Workspace';
+  } else if (scheme === 'jobs') {
+    sectionName = 'Jobs';
+  } else {
+    sectionName = displayName;
+  }
+
+  const applyFilter = (newFilter) => {
+    const prevQuery = queryString.parse(location.search);
+    const updatedQuery = queryString.stringify({
+      ...prevQuery,
+      filter: newFilter || undefined,
+      page: !infiniteScroll ? 1 : undefined,
+    });
+    history.push(`${location.pathname}?${updatedQuery}`);
+  };
+
+  const routeSearch = () => {
+    const prevQuery = queryString.parse(location.search);
+    const updatedQuery = queryString.stringify({
+      ...prevQuery,
+      query_string: query || undefined,
+      page: !infiniteScroll ? 1 : undefined,
+    });
+    history.push(`${location.pathname}?${updatedQuery}`);
+
+    if (siteSearch) window.dispatchEvent(new Event('portal.search'));
+  };
+
+  // Reset form field on route change
+  useEffect(() => {
+    !hasQuery && !siteSearch && setQuery('');
+  }, [hasQuery, location, siteSearch]);
+
+  const onSubmit = (e) => {
+    routeSearch();
+    e.preventDefault();
+  };
+
+  const onClear = (e) => {
+    e.preventDefault();
+    if (filterType && filterType !== 'All Types') {
+      applyFilter(undefined);
+    }
+    setQuery('');
+    history.push(location.pathname);
+  };
+
+  const onChange = (e) => setQuery(e.target.value);
+
+  return (
+    <form
+      aria-label={`${sectionName} Search`}
+      className={`${className} ${styles['container']}`}
+      onSubmit={onSubmit}
+    >
+      <div className={`input-group ${styles['query-fieldset']}`}>
+        <div className="input-group-prepend">
+          <Button
+            attr="submit"
+            type="secondary"
+            disabled={disabled}
+            size="medium"
+            iconNameBefore="search"
+          >
+            Search
+          </Button>
+        </div>
+        <input
+          type="search"
+          minLength="3"
+          onInput={(e) => e.target.setCustomValidity('')}
+          onInvalid={(e) =>
+            e.target.setCustomValidity(
+              'Include at least 3 characters in your search.'
+            )
+          }
+          onChange={onChange}
+          value={query || ''}
+          name="query"
+          aria-label={`Search ${sectionName}`}
+          className={`form-control ${styles['input']}`}
+          placeholder={`Search ${sectionName}`}
+          data-testid="input"
+          autoComplete="off"
+          disabled={disabled}
+        />
+      </div>
+
+      {filterTypes && filterTypes.length > 0 && api === 'tapis' && (
+        <div className={styles['file-filter']}>
+          <span>Filter:</span>
```

Comment:
My gut says:
* If it is currently only ever used for File Type, retain that text.
* If it is currently used for something else also, use "Filter".
* [If it may be in the future, Design would know (so I asked).](https://tacc-team.slack.com/archives/CQUA4D5KJ/p1668098753552879)

Reply:
[You may use the text "Filter". If you do, then also update the text "All Files" to be "All File Types".](https://tacc-team.slack.com/archives/CQUA4D5KJ/p1668113261194169?thread_ts=1668098753.552879&cid=CQUA4D5KJ)